### DISCUSSION TITLE
-vf reverse filter used 200GB of memory and still would not start.

### DISCUSSION THREAD

**Discussion by MaryAtlas** [nagadomi/nunif#239] [Q&A]
Hello.
On iw3 gui, I tried to use the "reverse" filter and got MemoryError. I tried to increase my Virtual Memory to 200GB and it ran for 5minutes, consistently increasing usage on the Virtual Memory until it even exceeded 200GB, but was still stuck at Loading "Depth Model" throughout.
There was no problem if I run the filter on cmdprompt only with ffmpeg. Please help.

Here are the screenshots:
![Capture](https://github.com/user-attachments/assets/65876652-cfe7-48cc-8a28-6b34e2dad2cf)
![Capture2](https://github.com/user-attachments/assets/15f95ab8-a21f-49c5-afc3-56fe021cbccd)

**Comment by MaryAtlas** [comment]
I want to add that, the input video was 4 seconds long.

**Comment by nagadomi** [comment]
The problem of an infinite loop when estimating output size has been fixed.
After updating, you should get an “Unable to estimate output size of video filter” error.

Regarding `reverse` filter, I do not plan to support it in iw3.
iw3 does not support audio filters, so it cannot support filters that change playback speed or position.
Use different software to edit the source video.

**Reply by MaryAtlas** [reply]
You are confused. reverse filter does not do anything to the audio. "reverse" filter changes the video, while "areverse" filter changes the audio. I am only using the "reverse" filter.

**Reply by nagadomi** [reply]
reverse video only will cause mismatch with audio.
also, reverse filter probably needs to load all the frames, which cannot handle huge videos.
there is no point in supporting such an incomplete feature.
